react/Empty/index.jsx

```diff
@@ -19,8 +19,14 @@ export const Empty = ({
   ...restProps
 }) => {
   if (layout) {
-    depreciationLogger(`Please add layout=false in the props of the Empty component, and implement the vertical
-    centering with a custom layout component. The use of Empty without layout=false is deprecated.`)
+    depreciationLogger(
+      `"layout" prop on Empty component is deprecated. Please add a style to the parent component for manage the position of the empty component and remain compatible with future major versions of cozy-ui`
+    )
+  }
+  if (layout === undefined) {
+    depreciationLogger(
+      `"layout" prop on Empty component is deprecated. For keep the margin top of Empty component (on Mobile), you can set "layout" prop to "true" but prefer add a style to the parent component for manage the position of the empty component and remain compatible with future major versions of cozy-ui`
+    )
```

Comment:
♻️ puisque layout est déprécié, autant proposer la solution finale.

Reply:
C'est juste, mais j'avais fait ainsi suite à cette issue: https://github.com/cozy/cozy-ui/issues/1896
Qu'en penses-tu :question: :slightly_smiling_face:

Reply:
🤔 je ne comprends plus trop ce qu'on cherche à faire...

De ce que j'ai compris : 
- Grosso modo la majorité des cas dans nos app ont layout undefined. Donc on forçait layout à true via les default props, donc il n'y a pas de centrage vertical.
- pour le cas dans Banks (et les autres à la marge donc) qui utilise layout false, il faudra, suite au BC, faire le centrage vertical dans l'app

Du coup si dans les app on forçait layout true, ça ne changeait rien, la seul manière de ne pas avoir le comportement par défaut était d'avoir layout false.

Or ici ce qu'on cherche à faire c'est supprimer ce comportement de centrage vertical.

Donc le cas impacté, et le seul, c'est pour le layout false. Donc il suffit de gérer if (layout === false) pour afficher le message de dépréciation, en disant genre "attention ce n'est plus supporté, vous devez gérer le centrage vertical via un wrapper"

Non ?

Reply:
Aujourd'hui, `layout={true}` ou non défini == "centrage" vertical

>Grosso modo la majorité des cas dans nos app ont layout undefined. Pour ces cas on force layout à true via les default props, donc il n'y a pas de centrage vertical.

C'est l'inverse 🙂 

>Pour les cas de layout true dans les apps, aucun changement, cela voulait juste dire qu'il spécifié le fait de ne pas centrer verticalement

Pas de changement visuel, mais elle auront ce warning https://github.com/cozy/cozy-ui/blob/0888d85092690b4f091aedde842ab7cd93dab916/react/Empty/index.jsx#L21-L25

>Pour le cas dans Banks (et les autres éventuels) qui utilise layout false, il faudra, suite au BC, faire le centrage vertical dans l'app

Là il n'y aura aucun changement et aucun warning 👍 

L'idée générale (voir l'issue lié), est de faire une migration en 2 temps, permettant aux apps ne spécifiant pas de `layout` de pouvoir le set à true si besoin de le fix très rapidement.

Et prochainement on pourra l'enlever complètement 🙂

Reply:
Exact ! Vu en zoom ;)